appserver/security/realm-stores/src/main/java/fish/payara/security/realm/RealmUtil.java

```diff
@@ -107,15 +106,25 @@ static void createAuthRealm(String name, String realmClass, String loginModule,
         EmbeddedSystemAdministrator administrator = serviceLocator.getService(EmbeddedSystemAdministrator.class);
         ParameterMap parameters = new ParameterMap();
         parameters.insert("authrealmname", name);
-        parameters.insert("property", props.entrySet().stream().map(Object::toString).collect(joining(":")));
+        parameters.insert("property", props.entrySet()
+                .stream()
+                .map(prop -> escapeRealmProperty(prop.getKey().toString()) + '=' + escapeRealmProperty(prop.getValue().toString()))
+                .collect(joining(":"))
+        );
         parameters.insert("classname", realmClass);
         if (loginModule != null) {
             parameters.insert("login-module", loginModule);
         }
         ActionReport report = new PlainTextActionReporter();
         ActionReport outreport = commandRunnerHelper.runCommand("create-auth-realm", parameters, report, administrator.getSubject());
-        if (outreport.getActionExitCode() != ActionReport.ExitCode.SUCCESS) {
-            throw new IdentityStoreException("Error in creating Auth realm: " + name);
+        if (outreport.getActionExitCode() == ActionReport.ExitCode.FAILURE) {
+            throw new IllegalStateException("Error in creating Auth realm: " + name);
         }
     }
+    
+    static String escapeRealmProperty(String component) {
+        return component
+                .replaceAll("\\:", "\\\\:")
+                .replaceAll("\\=", "\\\\=");
```

Comment:
Are you sure no handling is needed for backslash? so when my password is `a\password`, won't it break?

Reply:
Yes, I have checked, the property attribute only requires handling for colon and equal character.